### PR TITLE
Introduce labels in Log tab for commits that contain refactorings

### LINKED ISSUE #9: Label the commits that contain refactorings in the Log tab
# Summary
It would be great to label the commits that contain refactorings.
Currently, the user can't identify the commits that contain refactorings in the Log tab.

# Checklist
- [x] Manipulate existing VcsLogGraphTable
- [x] Mark commits with refactorings
- [x] Use Icon for marking
- [ ] Mark unmined commits with different icon
- [x] Add toggle to hide labels from vcs table
- [ ] (If possible) Align Icon on the right side of the table

### PR DESCRIPTION
**Pull request by bbrockbernd**
closes #9 

###  Labels shown in VCS table:
image

### Can be enabled or disabled in the presentation settings:
image

### CODE REVIEW

File: src/main/java/org/jetbrains/research/refactorinsight/actions/ToggleLabelsAction.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.ToggleAction;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.research.refactorinsight.services.WindowService;
+
+/**
+ * Toggle labeling of commits containing refactorings in VCSTable.
+ */
+public class ToggleLabelsAction extends ToggleAction {
```
Comment (onewhl) [inline src/main/java/org/jetbrains/research/refactorinsight/actions/ToggleLabelsAction.java:11]: Why not use `BooleanPropertyToggleAction` or `DumbAwareToggleAction`?

File: src/main/java/org/jetbrains/research/refactorinsight/services/WindowService.java [hunk 2]
```diff
@@ -35,15 +37,39 @@ public static WindowService getInstance(@NotNull Project project) {
    * @param state true for selected, false for unselected
    */
   public void setSelected(@NotNull MainVcsLogUi ui, boolean state) {
-    GitWindow gitWindow = gitInfo.computeIfAbsent(ui.getTable(), table -> {
-      Disposer.register(ui, () -> gitInfo.remove(ui.getTable()));
-      return new GitWindow(project, ui);
-    });
+    GitWindow gitWindow = gitInfo.get(ui.getTable());
```
Comment (onewhl) [inline src/main/java/org/jetbrains/research/refactorinsight/services/WindowService.java:40]: Could you please clarify this change?

File: src/main/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -45,6 +45,12 @@
             <add-to-group group-id="Vcs.Log.ChangesBrowser.Toolbar" anchor="first"/>
         </action>
 
+        <action id="actions.ToggleLabelsAction"
+                class="org.jetbrains.research.refactorinsight.actions.ToggleLabelsAction"
+                text="Show Refactoring Labels" description="Adds the refactoring icon to commits containing refactorings">
+            <add-to-group group-id="Vcs.Log.PresentationSettings" anchor="first"/>
```
Comment (onewhl) [inline src/main/resources/META-INF/plugin.xml:51]: Could we place the toggle above all "Show <something>" actions in the list?

File: src/main/java/org/jetbrains/research/refactorinsight/services/WindowService.java (hunk 2 above)
Comment (bbrockbernd) [reply src/main/java/org/jetbrains/research/refactorinsight/services/WindowService.java:40]: This has been moved to the update function. This ensures labels can be viewed before the refactorings view is opened!

File: src/main/java/org/jetbrains/research/refactorinsight/actions/ToggleLabelsAction.java (hunk 1 above)
Comment (bbrockbernd) [reply src/main/java/org/jetbrains/research/refactorinsight/actions/ToggleLabelsAction.java]: Good call, changed it to dumbaware

File: src/main/resources/META-INF/plugin.xml (hunk 3 above)
Comment (bbrockbernd) [reply src/main/resources/META-INF/plugin.xml]: Done

**Pull request merged**